### DISCUSSION TITLE
Suggestion: Update query_factory to include support for Transactions or multiple statements

### DISCUSSION THREAD

**Discussion by lat9** [zencart/zencart#5769] [Ideas]
The current versions of the `queryFactory` class do not support MySql [multiple statements](http://php.net/manual/en/mysqli.quickstart.multiple-statement.php).

For a 'tight' integration, where it's necessary to update multiple tables to record a certain condition it would be preferable to perform a single database request rather than issuing the updates one-by-one.

**Comment by drbyte** [comment]
Interesting idea. 
I can see the value, particularly when transaction-like statement groups need to be executed.

I've not seen other frameworks do this, or at least haven't dug in to see their use-cases for/against it. Instead, many have adopted PDO and full on transaction statements, on InnoDB tables.

NOTE: To use MySQL "transaction" features, the affected tables need to use InnoDB.

**Comment by drbyte** [comment]
Related to (and will probably defer until implementing) #2132 (InnoDB and Foreign Keys)
